src/dns-test.js

```diff
@@ -293,19 +341,43 @@ async function sendQueries(nameservers_ipv4) {
  */
 function sendTelemetry(payload) {
     payload.measurementID = measurementID;
-    browser.telemetry.submitPing(TELEMETRY_TYPE, payload, TELEMETRY_OPTIONS);
+    console.log(payload);
+    // browser.telemetry.submitPing(TELEMETRY_TYPE, payload, TELEMETRY_OPTIONS);
+}
+
+async function fetchTest() {
+    // Unsure if we want to wrap the fetch in a try/catch. This would enable 
+    // us to handle network errors, in case that matters to us. I don't think 
+    // it does, though. I think all we want is to stop the measurement from 
+    // running if we don't get the response we expect. Also, if the fetch fails, 
+    // it doesn't make sense to send anything to telemetry, because that 
+    // may fail, too.
+    let response = await fetch("https://dnssec-experiment-moz.net/");
+    let responseOk = await response.ok;
+    let responseText = await response.text();
+    if (!(responseOk && responseText && responseText === "Hello, world!\n")) {
+        throw new Error(STUDY_ERROR_FETCH_NOT_MATCHED);
+    }
 }
 
 /**
  * Entry point for our measurements.
  */
-async function runMeasurement() {
-    // If we can't upload telemetry. don't run the addon
-    let canUpload = await browser.telemetry.canUpload();
-    if (!canUpload) {
-        return
+async function runMeasurement(details) {
+    // Now that we're here, stop listening to the captive portal
+    browser.captivePortal.onConnectivityAvailable.removeListener(runMeasurement);
```

Comment:
I suggest to move this to the listener, since you're also invoking `runMeasurement` without registering the listener first.

```
browser.captivePortal.onConnectivityAvailable.addListener(function listener(details) {
  browser.captivePortal.onConnectivityAvailable.removeListener(listener);
  runMeasurement(details);
});
```

Side note: `details` of the `onConnectivityAvailable` event is not the same as `details` returned by `getState()`.

Reply:
Thanks! Regarding `onConnectivityAvailable` and `getState()`, it is kind of awkward that I'm passing the `details` from both to `runMeasurement()`. Do you think it's a problem for the correctness of the add-on, or do you think I should clean up the code somehow so that I'm not checking for strings that won't be passed to `runMeasurement()`?

Reply:
Or were you just suggesting that I remove the check for `captiveStatus === clear` in `main()`, since `getState()` can't return that string?

Reply:
Moved the `removeListener()` call in 959dca8, per your suggestion

Reply:
Removed the check for 'clear' after the call to getState(), since that is only returned for onConnectivityAvailable.addListener().

Also added some comments to clarify the return values for each function. Addressed by 7c6a475.